config/config.go

```diff
@@ -0,0 +1,331 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	logging "github.com/op/go-logging"
+	"github.com/urfave/cli"
+	git "gopkg.in/src-d/go-git.v4"
+	yaml "gopkg.in/yaml.v2"
+)
+
+var configLogger = logging.MustGetLogger("config")
+
+type configFileV1 struct {
+	// config version
+	Version string
+
+	CLI struct {
+		// Upload configuration.
+		APIKey  string `yaml:"api_key"`
+		Server  string
+		Project string
+		Locator string
+	}
+	Analyze struct {
+		Modules []ModuleConfig
+	}
+}
+
+// ModuleType is an enumeration of supported build system types
+type ModuleType string
+
+const (
+	// Individual tools
+
+	// Bower is the module type for bower.io
+	Bower = ModuleType("bower")
+	// Composer is the module type for getcomposer.org
+	Composer = ModuleType("composer")
+	// Maven is the module type for maven.apache.org
+	Maven = ModuleType("maven")
+	// SBT is the module type for scala-sbt.org
+	SBT = ModuleType("sbt")
+
+	// Ecosystems where many tools behave similarly
+
+	// Ruby is the module type for Bundler (bundler.io)
+	Ruby = ModuleType("ruby")
+	// Nodejs is the module type for NPM (npmjs.org) and Yarn (yarnpkg.com)
+	Nodejs = ModuleType("nodejs")
+	// Golang is the module type for dep, glide, godep, govendor, vndr, and manual
+	// gopath vendoring
+	Golang = ModuleType("golang")
+
+	// VendoredArchives is a module type for archive formats (.tar, .rpm, .zip, etc...)
+	VendoredArchives = ModuleType("vendoredarchives")
+)
+
+// ModuleConfig defines a config for a builder's entry point
+type ModuleConfig struct {
+	Name string
+	Path string
+	Type ModuleType
+}
+
+func readConfigFile(path string) (configFileV1, error) {
+	if path != "" {
+		if _, err := os.Stat(path); err != nil {
+			return configFileV1{}, fmt.Errorf("invalid config file specified")
+		}
+		return ParseConfigFile(path)
+	}
+
+	_, err := os.Stat(".fossa.yml")
+	if err == nil {
+		return ParseConfigFile(".fossa.yml")
+	}
+
+	_, err = os.Stat(".fossa.yaml")
+	if err == nil {
+		return ParseConfigFile(".fossa.yaml")
+	}
+
+	return setDefaultValues(configFileV1{})
+}
+
+// WriteConfigFile writes a config state to yaml
+func WriteConfigFile(filename string, conf configFileV1) error {
+	return errors.New("Not Implemented")
```

Comment:
This should be more descriptive and follow the formatting conventions for error messages. A better message would be `"WriteConfigFile is not implemented"`

Reply:
This will be removed when fossa init is implemented